src/Analyzers/Core/CodeFixes/UseObjectInitializer/AbstractUseObjectInitializerCodeFixProvider.cs

```diff
@@ -76,11 +82,54 @@ protected override async Task FixAsync(
         var statement = objectCreation.FirstAncestorOrSelf<TStatementSyntax>();
         Contract.ThrowIfNull(statement);
 
-        var newStatement = GetNewStatement(statement, objectCreation, matches)
-            .WithAdditionalAnnotations(Formatter.Annotation);
+        var firstToken = objectCreation.GetFirstToken();
+        var formattingOptions = await document.GetSyntaxFormattingOptionsAsync(
+            this.SyntaxFormatting, cancellationToken).ConfigureAwait(false);
+        var indentationOptions = new IndentationOptions(formattingOptions);
+
+        var newStatement = GetNewStatement(statement, objectCreation, indentationOptions, matches).WithAdditionalAnnotations(Formatter.Annotation);
 
         editor.ReplaceNode(statement, newStatement);
         foreach (var match in matches)
             editor.RemoveNode(match.Statement, SyntaxRemoveOptions.KeepUnbalancedDirectives);
     }
+
+    protected TExpressionSyntax Indent(TExpressionSyntax expression, IndentationOptions indentationOptions)
+    {
+        var insertionText = indentationOptions.FormattingOptions.UseTabs
```

Comment:
So, perhaps my mental model of UseTab, TabSize, and IndentSize are wrong, but I thought that the indentation was always IndentSize spaces wide and that could use 1 or more tabs based on TabSize.

UseTabs = true
TabSize = 2
IndentSize = 4

each indent would be `\t\t`.

UseTabs = false
TabSize = 2
IndentSize = 4

each indent would be `\s\s\s\s`.

Reply:
you are correct.  this doesn't properly implement the desired indentation logic.  I will fixup.  Nice catch.